homeassistant/components/control4/light.py

```diff
@@ -185,13 +187,15 @@ def create_api_object(self):
     @property
     def is_on(self):
         """Return whether this light is on or off."""
-        return self.coordinator.data[self._idx]["value"] > 0
+        if self._is_dimmer:
+            return self.coordinator.data[self._idx][CONTROL4_DIMMER_VAR] > 0
+        return self.coordinator.data[self._idx][CONTROL4_NON_DIMMER_VAR] > 0
```

Comment:
This looks like an unrelated bugfix. It should be a separate PR.

Reply:
This is a refactor to make use the new version of the helper function "update variables for entity"  to prevent unnecessary redundant helper functions. I will include it in that new preliminary PR along with a version bump and the helper function change

Reply:
Please consider instead making multiple preliminary PR.
Having one big preliminary PR would defy the point...

Reply:
Then what should be included in multiple preliminary PRs? The first PR just has the dependency bump and const changes and the second PR contains the new function and changes in the light platform.

Reply:
Something like that - see https://developers.home-assistant.io/docs/review-process#creating-the-perfect-pr

Reply:
For the second preliminary PR should I wait for the first PR to be merged then introduce it

Reply:
If it depends on the first one - please wait until it is merged.
If it is independant, you can create it now.

Reply:
Resolved by https://github.com/home-assistant/core/pull/90234